apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/keyboard-navigation.tsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+/**
+ * Internal dependencies
+ */
+import useFocusHandler from './hooks/use-focus-handler';
+import useKeydownHandler from './hooks/use-keydown-handler';
+
+interface Props {
+	onMinimize: () => void;
+	onDismiss: ( target: string ) => () => void;
+	onNextCardProgression: () => void;
+	onPreviousCardProgression: () => void;
+	focusRef: React.MutableRefObject< null | HTMLElement >;
+	isMinimized: boolean;
+}
+
+const KeyboardNavigation: React.FunctionComponent< Props > = ( {
+	onMinimize,
+	onDismiss,
+	onNextCardProgression,
+	onPreviousCardProgression,
+	focusRef,
+	isMinimized,
+} ) => {
+	function ExpandedTourNav() {
+		useKeydownHandler( {
+			onEscape: onMinimize,
+			onArrowRight: onNextCardProgression,
+			onArrowLeft: onPreviousCardProgression,
+		} );
+		return null;
+	}
+
+	function MinimizedTourNav() {
+		useKeydownHandler( { onEscape: onDismiss( 'esc-key' ) } );
```

Comment:
@razvanpapadopol I think this is fine right? (these are used to record where the tour was closed from)